src/main/java/org/apache/commons/lang3/StringEscapeUtils.java

```diff
@@ -429,6 +429,65 @@ public StringEscapeUtils() {
       super();
     }
 
+    /**
+     * <p>Convenience wrapper for {@link java.lang.StringBuilder} providing escape methods.</p>
+     *
+     * <p>Example:</p>
+     * <pre>
+     * new Builder(ESCAPE_HTML4)
+     *      .append("&lt;p>")
+     *      .escape("This is paragraph 1 and special chars like &amp; get escaped.")
+     *      .append("&lt;/p>&lt;p>")
+     *      .escape("This is paragraph 2 &amp; more...")
+     *      .append("&lt;/p>")
+     *      .toString()
+     * </pre>
+     * 
+     * @since 3.6
+     */
+    public static class Builder {
+
+        private final StringBuilder sb;
```

Comment:
The problem with `Appendable` is that its `append` method can throw an `IOException`. That would make the builder more cumbersome to use. Of course it could just rethrow the `IOException` as a `RuntimeException`, but I do not know if this is o.k. for commons-lang.